mpfr 4.0.0: deprecated functions

## Issue Details

There are warnings while compiling cgal when using latest mpfr version 4.0.0 regarding function `mpfr_root`. In their latest changes (http://www.mpfr.org/mpfr-current/#changes) they report that 

> Functions mpfr_grandom and mpfr_root are deprecated and will be removed in a future release.

```
/cgal/Number_types/include/CGAL/GMP/Gmpfr_type.h:889:49: warning: 
‘int mpfr_root(mpfr_ptr, mpfr_srcptr, long unsigned int, mpfr_rnd_t)’ is deprecated [-Wdeprecated-declarations]
         mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
```

## Environment

Non-rolling distros still have version 3.1.4-1 in their repositories.

--- a/Number_types/include/CGAL/GMP/Gmpfi_type.h
+++ b/Number_types/include/CGAL/GMP/Gmpfi_type.h
@@ -618,8 +618,13 @@ Gmpfi Gmpfi::kthroot(int k,Gmpfi::Precision_type p)const{
         // MPFI does not provide k-th root functions
         CGAL_assertion(p>=MPFR_PREC_MIN&&p<=MPFR_PREC_MAX);
         Gmpfi result (0, p);
-        mpfr_rootn_ui(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);
-        mpfr_rootn_ui(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
+        #if(MPFR_VERSION_MAJOR < 4)
+            mpfr_root(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD)
+            mpfr_root(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
+        #else
+            mpfr_rootn_ui(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);
+            mpfr_rootn_ui(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
+        #endif
         *(result._left.fr()) = result._interval.left;
         *(result._right.fr()) = result._interval.right;
         return result;

--- a/Number_types/include/CGAL/GMP/Gmpfi_type.h
+++ b/Number_types/include/CGAL/GMP/Gmpfi_type.h
@@ -619,7 +619,7 @@ Gmpfi Gmpfi::kthroot(int k,Gmpfi::Precision_type p)const{
         CGAL_assertion(p>=MPFR_PREC_MIN&&p<=MPFR_PREC_MAX);
         Gmpfi result (0, p);
         #if(MPFR_VERSION_MAJOR < 4)
-            mpfr_root(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD)
+            mpfr_root(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);
             mpfr_root(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
         #else
             mpfr_rootn_ui(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);

--- a/Number_types/include/CGAL/GMP/Gmpfi_type.h
+++ b/Number_types/include/CGAL/GMP/Gmpfi_type.h
@@ -618,8 +618,8 @@ Gmpfi Gmpfi::kthroot(int k,Gmpfi::Precision_type p)const{
         // MPFI does not provide k-th root functions
         CGAL_assertion(p>=MPFR_PREC_MIN&&p<=MPFR_PREC_MAX);
         Gmpfi result (0, p);
-        mpfr_root(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);
-        mpfr_root(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
+        mpfr_rootn_ui(&(result.mpfi())->left, left_mpfr(), k,GMP_RNDD);
+        mpfr_rootn_ui(&(result.mpfi())->right,right_mpfr(),k,GMP_RNDU);
         *(result._left.fr()) = result._interval.left;
         *(result._right.fr()) = result._interval.right;
         return result;

--- a/Number_types/include/CGAL/GMP/Gmpfr_type.h
+++ b/Number_types/include/CGAL/GMP/Gmpfr_type.h
@@ -885,7 +885,11 @@ CGAL_GMPFR_ARITHMETIC_FUNCTION(cbrt,mpfr_cbrt)
 inline
 Gmpfr Gmpfr::kthroot(int k,std::float_round_style r)const{
         Gmpfr result(0,CGAL_GMPFR_MEMBER_PREC());
-        mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
+        #if(MPFR_VERSION_MAJOR < 4)
+            mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
+        #else
+            mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
+        #endif
         return result;
 }
 
@@ -895,7 +899,11 @@ Gmpfr Gmpfr::kthroot(int k,
                      std::float_round_style r)const{
         CGAL_assertion(p>=MPFR_PREC_MIN&&p<=MPFR_PREC_MAX);
         Gmpfr result(0,p);
-        mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
+        #if(MPFR_VERSION_MAJOR < 4)
+            mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
+        #else
+            mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
+        #endif
         return result;
 }
 

--- a/Number_types/include/CGAL/GMP/Gmpfr_type.h
+++ b/Number_types/include/CGAL/GMP/Gmpfr_type.h
@@ -885,7 +885,7 @@ CGAL_GMPFR_ARITHMETIC_FUNCTION(cbrt,mpfr_cbrt)
 inline
 Gmpfr Gmpfr::kthroot(int k,std::float_round_style r)const{
         Gmpfr result(0,CGAL_GMPFR_MEMBER_PREC());
-        mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
+        mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
         return result;
 }
 
@@ -895,7 +895,7 @@ Gmpfr Gmpfr::kthroot(int k,
                      std::float_round_style r)const{
         CGAL_assertion(p>=MPFR_PREC_MIN&&p<=MPFR_PREC_MAX);
         Gmpfr result(0,p);
-        mpfr_root(result.fr(),fr(),k,_gmp_rnd(r));
+        mpfr_rootn_ui(result.fr(),fr(),k,_gmp_rnd(r));
         return result;
 }
 

